tests/spec/time-flags.spec.js

```diff
@@ -104,3 +104,45 @@ describe('node times (atime, mtime, ctime) with mount flags', function() {
     });
   });
 });
+
+function memoryFS(flags, callback) {
+  var name = util.uniqueName();
+  return new Filer.FileSystem({
+    name: name,
+    flags: flags || [],
+    provider: new Filer.FileSystem.providers.Memory(name)
+  }, callback);
+}
+
+describe('fsPromises.update',function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+  var filename = '/dir/file';
+  var newfilename = filename + '1';
+  
+  it('should not update ctime when calling fs.rename() with NOCTIME', function() {
```

Comment:
This test is passing because it's ending before any of the code executes.  Asynchronous code like this is tricky to test, because you have to make sure your test harness waits until the code in question has finished.  There are a few ways to do this.  First, we sometimes use a callback (e.g., `done` param to our function here).  You don't have one, so the test is going to be looking for your test to return a `Promise`, or it will assume that it is synchronous, and just wait til the stack finishes, and then end and assume things passed.

In this case, you need to have your top-level test function return a Promise.  You do return on line 130, but you are three call stacks deep in async code at that point, and it won't propagate back up to the right stack frame.

Make this code use the new promises stuff is going to mean doing a pretty big rewrite of `createTree` (note: it doesn't exist on `fsPromises`, even though you are calling it there.